File: src/main/java/me/devsnox/battlekits/inventory/Frame.java
package me.devsnox.battlekits.inventory;

import me.devsnox.battlekits.utils.ItemBuilder;
import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;

public abstract class Frame {

    private FrameOptions frameOptions;

    public Inventory inventory;

    public Frame(int size, String displayName, FrameOptions frameOptions) {
        this.frameOptions = frameOptions;
        this.inventory = Bukkit.createInventory(null, size, displayName);
    }

    public void render(Player player) {
        player.openInventory(this.inventory);
    }

    public void onClick(InventoryClickEvent event) {
        if(this.frameOptions.isNoItemMove()) {
            event.setCancelled(true);
        }
    }

    public void onClose(InventoryCloseEvent event) {}

    public void fill(ItemStack itemStack) {
        for(int i = 0; i < this.inventory.getSize(); i++) {
            this.inventory.setItem(i, itemStack);
        }
    }

    public void fill(int durability) {
        ItemBuilder itemBuilder = new ItemBuilder();

        itemBuilder.setType(Material.STAINED_GLASS_PANE);
        itemBuilder.setDisplayName(" ");
        itemBuilder.setDurability(7);

        for(int i = 0; i < this.inventory.getSize(); i++) {
            this.inventory.setItem(i, itemBuilder.build());
        }
    }
}

File: spigot/src/main/kotlin/me/devsnox/battlekits/inventory/Frame.kt
package me.devsnox.battlekits.inventory

import net.darkdevelopers.darkbedrock.darkness.spigot.builder.inventory.InventoryBuilder
import net.darkdevelopers.darkbedrock.darkness.spigot.builder.item.ItemBuilder
import net.darkdevelopers.darkbedrock.darkness.spigot.functions.events.cancel
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.inventory.InventoryCloseEvent
import org.bukkit.inventory.Inventory

@Suppress("LeakingThis")
abstract class Frame(
    size: Int,
    displayName: String,
    private val isNoItemMove: Boolean
) {

    var inventory: Inventory = InventoryBuilder(size, displayName).build()

    open fun render(player: Player) {
        player.openInventory(inventory)
    }

    open fun onClick(event: InventoryClickEvent) {
        if (isNoItemMove) event.cancel()
    }

    open fun onClose(event: InventoryCloseEvent) {}

    fun fill() {
        val itemBuilder = ItemBuilder(Material.STAINED_GLASS_PANE).setDisplayName(" ").setDurability(7)
        for (i in 0 until inventory.size) inventory.setItem(i, itemBuilder.build())
    }

    override fun toString(): String = "Frame(isNoItemMove=$isNoItemMove, inventory=${inventory.name})"

}
